lib/net.js

```diff
@@ -371,12 +371,14 @@ Socket.prototype.read = function(n) {
 };
 
 
-// FIXME(joyeecheung): this method is neither documented nor tested
-Socket.prototype.listen = function() {
-  debug('socket.listen');
-  this.on('connection', arguments[0]);
-  listenInCluster(this, null, null, null);
-};
+Socket.prototype.listen = internalUtil.deprecate(function () {
+    debug('socket.listen');
+    this.on('connection', arguments[0]);
+    listenInCluster(this, null, null, null);
+  },
```

Comment:
The deprecation code should be `DEP00XX`... the actual value would be assigned when the PR lands